src/main/java/com/sleekbyte/tailor/Tailor.java

```diff
@@ -278,6 +302,8 @@ private void analyzeFiles(Set<String> fileNames) throws CLIArgumentParserExcepti
         ConstructLengths constructLengths = configuration.parseConstructLengths();
         Set<Rules> enabledRules = configuration.getEnabledRules();
 
+        setNumberOfFilesBeforePurge();
```

Comment:
I think this could be simplified to just `numberOfFilesBeforePurge = configuration.numberOfFilesBeforePurge();` (no need to have a separate method for it) because that will default to `0`, which will be semantically identical to not providing the `--purge` flag.